crates/bevy_transform/src/systems.rs

```diff
@@ -2,76 +2,96 @@ use crate::components::{GlobalTransform, Transform};
 use bevy_ecs::{
     entity::Entity,
     query::{Changed, With, Without},
-    system::Query,
+    system::{Local, Query},
 };
 use bevy_hierarchy::{Children, Parent};
 
+/// Used for [`transform_propagate_system`]. Ignore otherwise.
+pub(crate) struct Pending {
+    parent: *const GlobalTransform,
+    changed: bool,
+    child: Entity,
+}
+
+// SAFE: Values are cleared after every frame and cannot otherwise be
```

Comment:
This safety comment doesn't make sense to me. You're describing why the *lifetime* is valid, but this comment is for why the pointer is thread-safe.